api/client/src/main/java/org/projectnessie/client/http/ResponseContext.java

```diff
@@ -25,10 +26,15 @@ public interface ResponseContext {
   /** Get the status of the response. */
   Status getStatus();
 
+  /**
+   * Get the input stream for the response. Returns null if the response did not contain any
+   * readable entity.
+   *
+   * @implSpec Calling this method multiple times should return the same input stream.
```

Comment:
This is especially true for `GZIPInputStream`: it's illegal to create 2 streams from the same non-compressed stream.